src/Mod/Arch/ArchWall.py

```diff
@@ -823,19 +826,72 @@ def getExtrusionData(self,obj):
                 return data
         length  = obj.Length.Value
 
-        # TODO currently layers were not supported when len(basewires) > 0
+        # TODO currently layers were not supported when len(basewires) > 0	##( or 1 ? )
         width = 0
-        widths = obj.OverrideWidth
-
-        if obj.OverrideWidth:
-            if obj.OverrideWidth[0]:
-                width = obj.OverrideWidth[0]
-        if not width:
-            if obj.Width:
-                width = obj.Width.Value
+
+        # Get width of each edge segment from Base Objects if they store it (Adding support in SketchFeaturePython, DWire...)
+        widths = []  # [] or None are both False
+        if obj.Base:
+            if hasattr(obj.Base, 'Proxy'):
+                if hasattr(obj.Base.Proxy, 'getWidths'):
+                    widths = obj.Base.Proxy.getWidths(obj.Base)  # return a list of Width corresponds to indexes of sorted edges of Sketch
+
+        # Get width of each edge/wall segment from ArchWall.OverrideWidth if Base Object does not provide it
+        if not widths:
+            if obj.OverrideWidth:
+                if Draft.getType(obj.Base) == 'Sketch':
+                    # If Base Object is ordinary Sketch, sort the width list in OverrrideWidth to correspond to indexes of sorted edges of Sketch
+                    try:
+                        import ArchSketchObject
+                    except:
+                        print("ArchSketchObject add-on module is not installed yet")
```

Comment:
What is this ArchSketchObject? If it provides new functionality you should submit it to the main FreeCAD code base, instead of keeping it as an add-on. Why should it be an add-on? Does it do something particularly complex that shouldn't be part of the main code?

Reply:
Thanks checking.

Yes, it is kind of experiment to add functions that intend to be put in an add-on.

Not to mention that I haven't checked how to do add-on, the functions are being drafted with my limited knowledge in coding that would be far from good enough to be reviewed and added in main code I think.

Just as @yorik does for BIM, think it would save main developer's time to do in add-on, and seem more easy to update the code for everybody in an add-on.

Reply:
Where is this ArchSketchObject module?

Reply:
Thanks for checking!

This is one of the piece of puzzle in the experiment that I have tested along with for a long while to make using sketch as base of Wall more usable.

Problem is this part has not been posted to the forum for users to test as I have not found the help document to do an add-on and have not cleaned up the usable code from the mess  -  the ArchSketchObject.py is kind of python exercise / fc experiment with mix of codes / remarks > 5000 lines testing adding bit by bit in a year.

One of the usage shown in demo videos posted is added commands to let user click on the edge of the sketch to edit its width / align.

Will single out the usable part and post on the forum bit by bit.

Reply:
[**FC Discussion :-**](https://forum.freecadweb.org/viewtopic.php?f=23&t=39060&p=354185#p354185)
OK, singled out some functions in my repository which do below:-

- GUI to edit / swap a Wall Segment (Sketch Edge) individual Align (Left, Right, Center cycle)

- GUI to edit a Wall Segment (Sketch Edge) individual Width

As shown in uploaded videos in earlier posts.  

https://github.com/paullee0/FreeCAD_SketchArch

Checking how to do Addon.